sklearn/metrics/ranking.py

```diff
@@ -292,7 +299,23 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
 
         fpr, tpr, tresholds = roc_curve(y_true, y_score,
                                         sample_weight=sample_weight)
-        return auc(fpr, tpr)
+        if max_fpr is None or max_fpr == 1:
+            return auc(fpr, tpr)
+
+        idx = np.where(fpr <= max_fpr)[0]
```

Comment:
fpr is increasing, so it would be clearer if we just did something like:
```
stop = np.searchsorted(fpr, max_fpr, 'right')
```
and below used `tpr[:stop]` instead of `tpr[idx]` and avoided having both names `idx_in` and `idx_out`